vulture_os/toolkit/api_parser/harfanglab/harfanglab.py

```diff
@@ -190,19 +190,27 @@ def format_log(self, log, log_type):
         return json.dumps(log)
 
     def execute(self):
-        since = self.last_api_call or (timezone.now() - timedelta(days=7))
-        to = timezone.now()
-
-        # delay the times of 2 minutes, to let the times at the API to have all logs
-        to = to - timedelta(minutes=2)
-
-        msg = f"Parser starting from {since} to {to}."
-        logger.info(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
-
         for log_type in ["alerts", "threats"]:
             index = 0
             total = 1
+            is_threshold_trigerred = False
+
             while index < total:
+                since = self.frontend.harfanglab_timestamps.get(f"since_{log_type}") or (timezone.now() - timedelta(days=7))
+                if not since.tzinfo:
+                    since = timezone.make_aware(since)
+                self.frontend.harfanglab_timestamps[f"since_{log_type}"] = since
+
+                to = self.frontend.harfanglab_timestamps.get(f"to_{log_type}") or (since + timedelta(hours=24))
+                if not to.tzinfo:
+                    to = timezone.make_aware(to)
+                self.frontend.harfanglab_timestamps[f"to_{log_type}"] = to
+
+                self.frontend.save()
```

Comment:
You should NOT update those values, as long as the collector didn't finish processing the actual logs!